src/dataCommand.ts

```diff
@@ -222,48 +222,26 @@ export abstract class DataCommand extends SfdxCommand {
   }
 
   /**
-   * Splits a sequence of 'key=value key="leftValue rightValue"   key=value'
-   * into a list of key=value pairs, paying attention to quoted whitespace.
+   * Splits a sequence of "key=value key=leftValue rightValue   key=value"
+   * into a list of key=value pairs, paying attention to whitespace in values.
+   *
+   * e.g. "key=value key=leftValue rightValue   key=value0"
+   * becomes ["key=value", "key=leftValue rightValue", "key=value"]
    *
    * This is NOT a full push down-automaton so do NOT expect full error handling/recovery.
    *
    * @param {string} text - The sequence to split
    */
   private parseKeyValueSequence(text: string): string[] {
-    const separator = /\s/;
-
-    let inSingleQuote = false;
-    let inDoubleQuote = false;
-    let currentToken: string[] = [];
-    const keyValuePairs: string[] = [];
-
-    const trimmedText = text.trim();
-    for (const currentChar of trimmedText) {
-      const isSeparator = separator.exec(currentChar);
-
-      if (currentChar === "'" && !inDoubleQuote) {
-        inSingleQuote = !inSingleQuote;
-        continue;
-      } else if (currentChar === '"' && !inSingleQuote) {
-        inDoubleQuote = !inDoubleQuote;
-        continue;
-      }
-
-      if (!inSingleQuote && !inDoubleQuote && isSeparator) {
-        if (currentToken.length > 0) {
-          keyValuePairs.push(currentToken.join(''));
-          currentToken = [];
-        }
-      } else {
-        currentToken.push(currentChar);
-      }
-    }
+    // The regex below uses a positive lookbehind and a positive lookahead to split the string into key=value pairs
+    //   regardless of whitespace or single quotes in the values
+    // Use regex101.com and paste this regex to see how it works
+    const matches = text.match(/(\S+)=(?<=['"]?)([\d\w'"\s]+)(?=\s|$)/g);
```

Comment:
@RodEsp the `\d` in `[\d\w'"\s]` is redundant, numbers are covered by `\w`

Reply:
I love all the red in the diff here :)

Reply:
I tried pasting into regex101, and it had errors?

you can save your regex101 like https://regex101.com/r/gYqdkb/1 (ctrl+s) might be nice to build out an example there and save the url in the comments

Reply:
@WillieRuemmele you need to pick the regex flavor. I chose ECMAScript and it worked for me.